browser/lib/transport/xhrpollingtransport.js

```diff
@@ -0,0 +1,37 @@
+var XHRPollingTransport = (function() {
+	var shortName = 'xhr_polling';
+
+	function XHRPollingTransport(connectionManager, auth, params) {
+		params.stream = false;
+		CometTransport.call(this, connectionManager, auth, params);
+		this.shortName = shortName;
+	}
+	Utils.inherits(XHRPollingTransport, CometTransport);
+
+	XHRPollingTransport.isAvailable = XHRRequest.isAvailable;
+	XHRPollingTransport.checkConnectivity = XHRStreamingTransport.checkConnectivity;
+
+	XHRPollingTransport.tryConnect = function(connectionManager, auth, params, callback) {
+		var transport = new XHRPollingTransport(connectionManager, auth, params);
+		var errorCb = function(err) { callback(err); };
+		transport.on('error', errorCb);
+		transport.on('preconnect', function() {
+			Logger.logAction(Logger.LOG_MINOR, 'XHRPollingTransport.tryConnect()', 'viable transport ' + transport);
+			transport.off('error', errorCb);
+			callback(null, transport);
+		});
+		transport.connect();
+	};
+
+	XHRPollingTransport.prototype.toString = function() {
+		return 'XHRPollingTransport; uri=' + this.baseUri + '; isConnected=' + this.isConnected;
+	};
+
+	XHRPollingTransport.prototype.createRequest = XHRRequest.createRequest;
+
+	if(typeof(ConnectionManager) !== 'undefined' && XHRPollingTransport.isAvailable()) {
```

Comment:
Not a clue. Not my addition, it's been part of xhrtransport.js [since the initial commit](https://github.com/ably/ably-js/commit/8006e0c#diff-1b55013b80a5912a628e0c5b392b4d6cR36)